src/document/generator.ml

```diff
@@ -1357,9 +1363,10 @@ module Make (Syntax : SYNTAX) = struct
       Item.Declaration { attr; anchor; doc; content; source_anchor }
 
     and simple_expansion :
```

Comment:
This function should probably do something to the sidebar ? Currently, entering a sub-module do not change the sidebar. The current module isn't visible in the sidebar, which still highlight the parent module.

Reply:
Yes, but I propose to keep the current PR with the minimalest global sidebar: only toplevel entries.

It is very likely that we want to expand the sidebar, but I think that belongs to another PR (which will likely contain a lot of discussion).

Reply:
I do find it very awkward to be threading the sidebar through all these calls in `document/generator.ml` for very little benefit. It would seem sensible to me to see if we could instead pass a sidebar argument to the renderer; or at least set it in the pages after they've been generated.

Reply:
I removed the threading of the sidebar through `document/generator.ml` in [8d2c0cf](https://github.com/ocaml/odoc/pull/1145/commits/8d2c0cf969ed61578dd0020e6e49414e86c6d243).
A sidebar value is now passed to the renderer, which turns the sidebar into a document (depending on the page generated) and then pass that to the backend renderer.

Reply:
I am still not sure that it was not better to pass the sidebar around: the value can be used for different usecases, as for instance a hypothetical `{toc}` odoc syntax that would generate a table of content.